cmd/soroban-rpc/internal/db/ledgerentry.go

```diff
@@ -148,6 +196,29 @@ func (l *ledgerEntryReadTx) GetLedgerEntry(key xdr.LedgerKey) (bool, xdr.LedgerE
 	if err = xdr.SafeUnmarshal([]byte(ledgerEntryBin), &result); err != nil {
 		return false, xdr.LedgerEntry{}, err
 	}
+
+	// Disallow access to entries that have expired. Expiration includes the
+	// "current" ledger, which we are building.
+	// TODO: Support allowing access, but recording for simulateTransaction
+	switch result.Data.Type {
+	case xdr.LedgerEntryTypeContractData:
```

Comment:
fallthrough ?

Reply:
Are you suggesting a fallthrough? or?

Reply:
The codepath here seems to be identical other then the referenced field ContractData / ContractCode.
maybe something like:
```golang
expirationLedger := -1
switch result.Data.Type {
	case xdr.LedgerEntryTypeContractData:
		expirationLedger = result.Data.ContractData.ExpirationLedgerSeq
        case xdr.LedgerEntryTypeContractCode:
		expirationLedger = result.Data.ContractCode.ExpirationLedgerSeq
}
if expirationLedger > -1 {
       latestLedger, err := l.GetLatestLedgerSequence()
	if err != nil {
		return false, xdr.LedgerEntry{}, err
	}
        if expirationLedger <= xdr.Uint32(latestLedger) {
		return false, xdr.LedgerEntry{}, nil
	}
}
```